src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // eslint-disable-next-line max-len
+  const dateParts = date.split(fromFormat[3]); // Використовуємо роздільник з fromFormat
+
+  // Створюємо об'єкт для зручного доступу до частин дати
+  const dateObj = {};
+
+  fromFormat.slice(0, 3).forEach((format, index) => {
+    dateObj[format] = dateParts[index];
+  });
```

Comment:
It's a good practice to validate that `dateParts[index]` exists before assigning it to `dateObj[format]`. This prevents potential `undefined` assignments if the input structure changes unexpectedly, although the input is assumed to be correct as per the task.